### PR TITLE
Update OLCNE to version 1.2

### PR DESCRIPTION
**Pull request by mark-au**
This updates the script to install OLCNE 1.2 and removes the multi-nic workarounds no longer necessary in 1.2.

Tested:
* One control plane node, two compute node
* Three control plane node, two compute node

Signed-off-by: Mark Cram <mark.cram@oracle.com>

To prove mutli-nic workarounds no longer necessary (i.e. internal 192.168.99.0/24 addresses used):
Before changes:
```
[vagrant@master1 ~]$ k get nodes -o wide
NAME                 STATUS   ROLES    AGE   VERSION             INTERNAL-IP      EXTERNAL-IP   OS-IMAGE                  KERNEL-VERSION                      CONTAINER-RUNTIME
master1.vagrant.vm   Ready    master   42m   v1.17.9+1.0.5.el7   192.168.99.101   <none>        Oracle Linux Server 7.9   5.4.17-2036.100.6.1.el7uek.x86_64   cri-o://1.17.0
master2.vagrant.vm   Ready    master   44m   v1.17.9+1.0.5.el7   192.168.99.102   <none>        Oracle Linux Server 7.9   5.4.17-2036.100.6.1.el7uek.x86_64   cri-o://1.17.0
master3.vagrant.vm   Ready    master   45m   v1.17.9+1.0.5.el7   192.168.99.103   <none>        Oracle Linux Server 7.9   5.4.17-2036.100.6.1.el7uek.x86_64   cri-o://1.17.0
worker1.vagrant.vm   Ready    <none>   42m   v1.17.9+1.0.5.el7   192.168.99.111   <none>        Oracle Linux Server 7.9   5.4.17-2036.100.6.1.el7uek.x86_64   cri-o://1.17.0
worker2.vagrant.vm   Ready    <none>   41m   v1.17.9+1.0.5.el7   192.168.99.112   <none>        Oracle Linux Server 7.9   5.4.17-2036.100.6.1.el7uek.x86_64   cri-o://1.17.0
```

After changes:
```
[vagrant@master1 ~]$ k get nodes -o wide
NAME                 STATUS   ROLES    AGE     VERSION          INTERNAL-IP      EXTERNAL-IP   OS-IMAGE                  KERNEL-VERSION                      CONTAINER-RUNTIME
master1.vagrant.vm   Ready    master   6m21s   v1.18.10+2.el7   192.168.99.101   <none>        Oracle Linux Server 7.9   5.4.17-2036.100.6.1.el7uek.x86_64   cri-o://1.18.3
master2.vagrant.vm   Ready    master   8m2s    v1.18.10+2.el7   192.168.99.102   <none>        Oracle Linux Server 7.9   5.4.17-2036.100.6.1.el7uek.x86_64   cri-o://1.18.3
master3.vagrant.vm   Ready    master   10m     v1.18.10+2.el7   192.168.99.103   <none>        Oracle Linux Server 7.9   5.4.17-2036.100.6.1.el7uek.x86_64   cri-o://1.18.3
worker1.vagrant.vm   Ready    <none>   10m     v1.18.10+2.el7   192.168.99.111   <none>        Oracle Linux Server 7.9   5.4.17-2036.100.6.1.el7uek.x86_64   cri-o://1.18.3
worker2.vagrant.vm   Ready    <none>   9m48s   v1.18.10+2.el7   192.168.99.112   <none>        Oracle Linux Server 7.9   5.4.17-2036.100.6.1.el7uek.x86_64   cri-o://1.18.3
```

### CODE REVIEW

File: OLCNE/scripts/provision.sh [hunk 1]
```diff
@@ -247,7 +247,7 @@ setup_repos() {
   # Add OLCNE 1.1 channel
   echo_do yum install -y oracle-olcne-release-el7
   echo_do yum-config-manager --disable ol7_olcne
-  echo_do yum-config-manager --enable ol7_olcne11
+  echo_do yum-config-manager --enable ol7_olcne12
```
Comment (Djelibeybi) [inline OLCNE/scripts/provision.sh:250]: Do we not need to specifically `--disable ol7_olcne11` as well?

**Review by Djelibeybi — CHANGES_REQUESTED** [review]
Would it make sense to change the nodes to OL8 for 1.2?

**Comment by mark-au** [comment]
> Would it make sense to change the nodes to OL8 for 1.2?

I didn't think of that, will do.

**Comment by Djelibeybi** [comment]
> I didn't think of that, will do.

This has the handy side-effect of not requiring the disabling of older OLCNE repos as there aren't any for OL8. :)

**Comment by Djelibeybi** [comment]
We should also remove the `Kubernetes/` project completely as that is no longer supported by Oracle.

**Comment by Djelibeybi** [comment]
You also need to change the copyright year to 2021. 😛

File: OLCNE/Vagrantfile [hunk 2]
```diff
@@ -105,7 +105,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   # Use specific component version (mainly used for development)
   OLCNE_VERSION = default_s('OLCNE_VERSION', '')
   K8S_VERSION = default_s('K8S_VERSION', '')
```
Comment (AmedeeBulle) [inline OLCNE/Vagrantfile:107]: To be consistent, we can drop the K8S_VERSION from the Vagrantfile as well

**Review by AmedeeBulle — COMMENTED** [review]
When provisioning the operator you get:
```
    operator: Warning: There is another connection with the name 'System eth1'. Reference the connection by its uuid '...'
```

I believe we need to add `auto_config: false` to `operator.vm.network "private_network", ip: "192.168.99.100"`in the Vagrantfile

File: OLCNE/Vagrantfile (hunk 2 above)
Comment (Djelibeybi) [reply OLCNE/Vagrantfile:107]: Yep, good catch.

**Comment by Djelibeybi** [comment]
> I believe we need to add `auto_config: false` to `operator.vm.network "private_network", ip: "192.168.99.100"`in the Vagrantfile

Or we remove `auto_config: false` from everything else and just let Vagrant configure things itself. No need for us to do it manually.

**Comment by AmedeeBulle** [comment]
> > I believe we need to add `auto_config: false` to `operator.vm.network "private_network", ip: "192.168.99.100"`in the Vagrantfile
> 
> Or we remove `auto_config: false` from everything else and just let Vagrant configure things itself. No need for us to do it manually.

To be honest, I can't remember why we initially did the override in the script. There was something which wasn't setup properly at that time. I'll double check tomorrow to be sure we don't introduce a regression.

**Comment by AmedeeBulle** [comment]
We had to disable auto_config to be able to tweak the routing in previous OLCNE versions. So we should be good now withe the defaults...
I'll run my usual tests tomorrow to finalize the review

**Comment by Djelibeybi** [comment]
> I'll run my usual tests tomorrow to finalize the review

Perfect, thanks!

**Comment by mark-au** [comment]
I've got a fix for the kubectl-proxy changes that I'm testing so don't merge this one just yet

**Comment by Djelibeybi** [comment]
I've switched it to draft. When you've committed your `kubectl-proxy` changes, you can switch it back.

**Review by AmedeeBulle — COMMENTED** [review]
Verified the network setup, no issue there.

Ran some tests, cluster come nicely up, with all pods tight to the right IP addresses, so it looks good to me.

Good job @mark-au 

Side note: I am still getting too often timeouts while deploying (istio) modules -- it is a pity because it aborts the overall Vagrant deployment and you have to re-start from scratch...
It would be nice if we could bump up that 5 minutes timeout...

**Comment by Djelibeybi** [comment]
> Side note: I am still getting too often timeouts while deploying (istio) modules -- it is a pity because it aborts the overall Vagrant deployment and you have to re-start from scratch...
> It would be nice if we could bump up that 5 minutes timeout...

Yes, it would. In the interim, we should probably call out the Registry project and provide/highlight/call attention to the instructions on how to sync the images to a local registry and configure Vagrant to deploy OLCNE using that local registry.

**Review by Djelibeybi — APPROVED** [review]

**Pull request merged**